pkg/beacon/relay/dkg/dkg_test.go

```diff
@@ -0,0 +1,157 @@
+package dkg
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	bn256 "github.com/ethereum/go-ethereum/crypto/bn256/cloudflare"
+	"github.com/keep-network/keep-core/pkg/beacon/relay/event"
+	"github.com/keep-network/keep-core/pkg/beacon/relay/gjkr"
+	"github.com/keep-network/keep-core/pkg/beacon/relay/group"
+	"github.com/keep-network/keep-core/pkg/chain"
+	"github.com/keep-network/keep-core/pkg/chain/local"
+)
+
+var (
+	playerIndex                 group.MemberIndex
+	groupPublicKey              *bn256.G2
+	gjkrResult                  *gjkr.Result
+	dkgResultChannel            chan *event.DKGResultSubmission
+	startPublicationBlockHeight uint64
+	localChain                  chain.Handle
+	blockCounter                chain.BlockCounter
+)
+
+func setup() {
+	playerIndex = group.MemberIndex(1)
+	groupPublicKey = new(bn256.G2).ScalarBaseMult(big.NewInt(10))
+	gjkrResult = &gjkr.Result{GroupPublicKey: groupPublicKey}
+	dkgResultChannel = make(chan *event.DKGResultSubmission)
```

Comment:
We can add `1` element buffer here and tests can be simplified by removing the goroutine and block mining:
```
dkgResultChannel = make(chan *event.DKGResultSubmission, 1)
```
and later:
```
func TestDecideMemberFate_HappyPath(t *testing.T) {
	setup()

	dkgResultChannel <- &event.DKGResultSubmission{
		(...)
	}

	err := decideMemberFate(
		(...)
	)

	(...)
}
```